crates/sui-core/src/node_sync/node_follower.rs

```diff
@@ -534,6 +597,52 @@ where
             error: e.to_string(),
         })?
     }
+
+    pub fn sync_checkpoint(
+        &self,
+        checkpoint_contents: &CheckpointContents,
+    ) -> impl Stream<Item = SuiResult> {
+        let futures: FuturesOrdered<_> = checkpoint_contents
+            .iter()
+            .map(|digests| {
```

Comment:
Honesty I would make the base case the case where we send and process the whole block of transaction, and the special case the case where we have a block of 1.